src/editors/containers/TextEditor/components/SelectImageModal/index.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { Button, Stack } from '@edx/paragon';
+import { Add } from '@edx/paragon/icons';
+
+import { thunkActions } from '../../../../data/redux';
+import hooks from './hooks';
+import BaseModal from '../BaseModal';
+import ErrorAlert from './ErrorAlert';
+import SearchSort from './SearchSort';
+import Gallery from './Gallery';
+
+export const SelectImageModal = ({
+  fetchImages,
+  uploadImage,
+  isOpen,
+  close,
+  setSelection,
+}) => {
+  const {
+    imgList,
+    searchString, setSearchString,
+    sortFilter, setSortFilter,
+    highlighted, setHighlighted,
+    addFileRef,
+    addFileClick,
+    addFile,
+    onConfirmSelection,
+    loading,
+    error, setError,
+  } = hooks.imgHooks({ fetchImages, uploadImage, setSelection });
+
+  return (
+    <BaseModal
+      close={close}
+      confirmAction={(
+        <Button
+          variant="primary"
+          onClick={onConfirmSelection}
+          disabled={!highlighted}
+        >
+          Next
+        </Button>
+      )}
+      handleUpload={addFileClick}
+      isOpen={isOpen}
+      footerAction={(
+        <Button iconBefore={Add} onClick={addFileClick} variant="link">
+          Upload a new image
+        </Button>
+      )}
+      showUploadButton
+      title="Add an image"
+    >
+      <ErrorAlert
```

Comment:
this needs to be a method returned by the hook, so that it can be mocked for the snapshot.